File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        # num_set = set((nums))
        # return len(num_set) != len(nums)
        
        # num_set = set(())
        # for i in range(len(nums)):
        #     num_set.add(nums[i])
        #     if len(num_set) == i:
        #         return True
        # return False

        num_set = set(())
        for num in nums:
            if num in num_set:
                return True
            num_set.add(num)
        return False

File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        int checks = 0;
        if(k >= nums.length){
            checks = 1;
        } else{
            checks = nums.length - k + 1;
        }
        System.out.println(checks);
        Set<Integer> num_set = new HashSet<>();
        for(int num : nums){
            num_set.add(num);
        }
        if(num_set.size() == nums.length){
            return false;
        }

        List<Integer> prev_ints = new ArrayList<>();
        for(int i=0; i<checks; i++){
            
            for(int j=i; j<=i+k && j<nums.length; j++){
                if(prev_ints.contains(nums[j])){
                    return true;
                } else{
                    prev_ints.add(nums[j]);
                }
            }
            prev_ints.clear();
        }
        return false;
    }
}